src/dune_cache_storage/util.ml

```diff
@@ -35,6 +29,10 @@ let add_atomically ~mode ~src ~dst : Write_result.t =
     (match Optimistically.link ~src ~dst with
      | () -> Ok
      | exception Unix.Unix_error (Unix.EEXIST, _, _) -> Already_present
+     | exception Unix.Unix_error (Unix.EMLINK, _, _) ->
```

Comment:
Btw, I think this bit is actually unnecessary because the only caller passes a fresh temp file as `src`. I didn't import this bit internally. (And according to https://github.com/ocaml/dune/pull/8816 this function isn't used anywhere else.)

Reply:
Added a comment in the code to reflect this.